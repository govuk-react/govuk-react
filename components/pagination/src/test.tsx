import * as React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';

import { Pagination } from '.';

describe('Pagination', () => {
  const wrapper = (
    <Pagination>
      <Pagination.Anchor href="#">Example</Pagination.Anchor>
    </Pagination>
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(wrapper, div);
  });

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });
});
