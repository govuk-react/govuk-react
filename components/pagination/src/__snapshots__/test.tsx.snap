// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination matches wrapper snapshot: wrapper mount 1`] = `
.c1 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  list-style-type: none;
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  text-transform: none;
  font-size: 20px;
  line-height: 1.25;
  width: 100%;
}

.c1 > a {
  box-sizing: border-box;
  color: #1d70b8;
  fill: currentColor;
  padding: 15px;
  background-color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c1 > a:hover {
  background-color: #f8f8f8;
}

.c1 > a:focus {
  outline: 3px solid #ffdd00;
}

.c1 > a:visited {
  color: #2e358b;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 25px;
  padding: 10px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  margin-bottom: 20px;
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 27px;
    line-height: 1.25;
  }
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    margin-bottom: 30px;
  }
}

<Pagination>
  <src__StyledList>
    <ul
      className="c0"
    >
      <PaginationAnchor
        href="#"
      >
        <anchor__PaginationWrapper>
          <li
            className="c1"
          >
            <anchor__Anchor
              href="#"
            >
              <a
                className="anchor__Anchor-sc-2udijf-3"
                href="#"
              >
                <anchor__InnerWrap>
                  <div
                    className="c2"
                  >
                    Example
                  </div>
                </anchor__InnerWrap>
              </a>
            </anchor__Anchor>
          </li>
        </anchor__PaginationWrapper>
      </PaginationAnchor>
    </ul>
  </src__StyledList>
</Pagination>
`;
